v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/PubSubToElasticsearch.java

```diff
@@ -153,16 +154,16 @@ public static PipelineResult run(PubSubToElasticsearchOptions options) {
                 .build());
 
     /*
-     * Step 3b: Write elements that failed processing to error output table via {@link BigQueryIO}.
+     * Step 3b: Write elements that failed processing to deadletter PubSub topic via {@link PubSubIO}.
      */
-    convertedPubsubMessages
-        .get(TRANSFORM_DEADLETTER_OUT)
-        .apply(
-            "WriteTransformFailuresToBigQuery",
-            ErrorConverters.WritePubsubMessageErrors.newBuilder()
-                .setErrorRecordsTable(options.getErrorOutputTable())
-                .setErrorRecordsTableSchema(SchemaUtils.DEADLETTER_SCHEMA)
-                .build());
+    if (StringUtils.isNotBlank(options.getErrorOutputTopic())) {
```

Comment:
Since Pub/Sub topic is mandatory parameter and cannot be blank based on regex check adding if clause doesn't add any value.  Better to remove the if clause